datrie/meson.build

```diff
@@ -0,0 +1,77 @@
+ldflags = []
+link_depends = ['libdatrie.map']
+
+if ld_has_version_script
+    ldflags += [
+        '-Wl,--version-script,@0@'.format(meson.current_source_dir() / 'libdatrie.map')
+    ]
+endif
+
+if compiler.get_id() == 'msvc'
+    link_def = custom_target(
+        'link-def',
+        input : ['libdatrie.def'],
+        command : [powershell, '-Command', '''& {
+            Write-Output "EXPORTS" > @OUTPUT0@
+            Get-Content -Path @INPUT0@ >> @OUTPUT0@
+        }'''],
+        output : 'libdatrie.def',
+    )
+    ldflags += [ '/def:@0@'.format(link_def.full_path()) ]
+    link_depends += [link_def]
+endif
+
+datrie_src = files(
+    'alpha-map-private.h',
+    'alpha-map.c',
+    'alpha-map.h',
+    'darray.c',
+    'darray.h',
+    'dstring-private.h',
+    'dstring.c',
+    'dstring.h',
+    'fileutils.c',
+    'fileutils.h',
+    'tail.c',
+    'tail.h',
+    'trie-private.h',
+    'trie-string.c',
+    'trie-string.h',
+    'trie.c',
+    'trie.h',
+    'triedefs.h',
+    'typedefs.h',
+)
+
+datrie = library(
+    'datrie', datrie_src,
+    
+    dependencies : deps,
+    install : true,
+    link_args : ldflags,
+    link_depends : link_depends,
+    version : meson.project_version(),
+    include_directories : configuration_inc,
+    vs_module_defs : 'libdatrie.def',
```

Comment:
This break Windows gcc

Reply:
The vs module defs file? Why?

Reply:
I thought it would supply the .def file generated in the block for msvc above (I think I forgot to change this line to reference the generated file). However, the original def file here got passed to gcc which breaks.

I'm not sure why - I'm not exactly familiar with msvc/mingw (and it seems this project doesn't run on Windows either) I'm guessing that both compilers use the same linker def format on Windows? Maybe if I supply the generated file to gcc it might work.

Reply:
Ah, right, you'll need the `link_def` variable reference which contains a custom_target output.

I'm not actually sure I understand the custom_target? Is it just prepending a single constant line? Why isn't this changed in the source file so that you don't need to generate anything?

Also yes, the GNU compiler implementation in meson passes the defs file to the linker as this is valid there.

Reply:
Yes it is passing a single line. I'm not sure how the original file is used, but it was used as-is in autotools (and I don't know which platform it was for - on Linux it use .map) so I didn't want to touch it. On Windows you need a section marker so I took inspiration from #15 to add it during build time.